architecture/adr-list/adr_002_identity_transactions.md

```diff
@@ -44,103 +44,191 @@ _**Note**: Hyperledger Indy also contains other transaction types beyond the one
+- DID Cheqd Method: the hardcoded string `cheqd:` indicating that the identifier uses this DID Method specification.
+- DID Cheqd Namespace: a string that identifies the name of the primary Cheqd ledger, followed by a `:`. The namespace string may optionally have a secondary ledger name prefixed by a `:` following the primary name. If there is no secondary ledger element, the DID resides on the primary ledger, else it resides on the secondary ledger. By convention, the primary is a production ledger while the secondary ledgers are non-production ledgers (e.g. main, test) associated with the primary ledger.
+- Namespace Identifier: a self-certifying identifier unique to the given DID Cheqd namespace. To be self-certifying the identifier must be derived from the initial verkey associated with the identifier.
+
+The components are assembled as follows:
+
+`did:cheqd:<namespace>:<namespace identifier>`
+
+Some examples of did:cheqd DID Method identifiers are:
+
+- A DID written to the Cheqd MainNet ledger:
+  `did:cheqd:main_net:7Tqg6BwSSWapxgUDm9KKgg`
+- A DID written to the Cheqd TestNet ledger:
+  `did:cheqd:test_net:6cgbu8ZPoWTnR5Rv5JcSMB`
+
+
 ### General structure of transaction requests
 
 All identity requests will have the following format:
 
 ```text
 {
     "data": { <request data for writing a transaction to the ledger> },
-    "owner": "GEzcdDLhCpGCYRHW82kjHd",
-    "signature": "49W5WP5jr7x1fZhtpAhHFbuUDqUYZ3AKht88gUjrz8TEJZr5MZUPjskpfBFdboLPZXKjbGjutoVascfKiMD5W7Ba",
+    "creators": [<identifier>, ...]
+    "signatures": [
+                    <public_key>: <signature>,
+                  ],
     "metadata": {}
 }
 ```
 
 * **`data`**: Data requested to be written to the ledger, specific for each request type.
-* **`owner`**: Owner identifier \(DID\) for this entity. This could be a new DID or an existing DID, for existing entities.
-* **`signature`**: `data` should be signed by `owner` private key.
+* **`creators`**: Creators identifiers (DID) list for this entity. There should be a new DIDs or an existing DIDs, for existing entities.
+* **`signatures`**: `data` should be signed by all `creators` private key. This field contains a dict there creator's public key is a key, and the signature is a value.
 * **`metadata`**: Dictionary with additional metadata fields. Empty for now. This fields provides extensibility in the future, e.g., it can contain `protocolVersion` or other relevant metadata associated with a request.
 
 ## List of transactions and details
 
 ### `DID` transactions
 
 [Decentralized Identifiers \(DIDs\) are a W3C specification](https://www.w3.org/TR/did-core/) for identifiers that enable verifiable, decentralized digital identity.
+DIDDoc format conforms to [DIDDoc spec]().
+The request can be used for creation of new DIDDoc, setting, and rotation of verification key.
 
-The request can be used for creation of new DIDs, setting, and rotation of verification key.
+#### DIDDoc
+1. **`id` (base58-encoded string):**
```

Comment:
I know this has been discussed in meetings, BUT my point is that if we implement the DID Doc and store them in the ledger, this document must conform to the DID Doc specification. I am not sure if it is correct to convert this structure to DID Doc in VDR tool. What happens if someone implements another client and suspects the ledger will return the DID Doc?

Reply:
My concerns:
- How are we going to validate this prefix? With hard-coded values?
- What are we going to do after changing this prefix? Is it possible?

Reply:
I guess if prefix has changed it means did method has changed as well, so it's okey for me hard-coded them.

Reply:
We should store the full DID, as in, `did:cheqd:network:identifier` in the body of this request. What the DID method is called and structure is up to us to define. If this is NOT on ledger, then the interpretation of which method is being used, what network, and what structure to expect is happening entirely on a backend/client application.

Not storing the full DID, as far as I understand, violates the DID Core specificiation.

Reply:
@Toktar What are your thoughts on this?

Reply:
@Toktar to check with Sergey K. DID Core spec I'm pretty sure needs fully qualified.